File: frontend/src/types/post.ts
export interface Media {
  id: string;
  name: string;
  type: 'image' | 'video' | 'document' | 'audio';
  mime_type: string;
  url: string;
  size: number;
  width?: number;
  height?: number;
  duration?: number;
  uploaded_at: string;
  post_id?: string;
  user_id: string;
  is_processed: boolean;
  is_public: boolean;
  virus_scan_status?: 'pending' | 'clean' | 'infected';
}

export interface Post {
  id: string;
  title: string;
  summary?: string;
  content?: string;
  specialty?: string;
  is_urgent: boolean;
  is_resolved: boolean;
  ai_suggest?: string;
  author_id: string;
  media: Media[];
  tags: string[];
  created_at: string;
  updated_at?: string;
  comment_count: number;
  like_count: number;
  solution_count: number;
  is_anonymized: boolean;
}

export interface PostWithAuthor extends Post {
  author?: {
    id: string;
    name: string;
    avatar?: string;
    specialty?: string;
  };
}

export interface PostCreate {
  title: string;
  summary?: string;
  content?: string;
  specialty?: string;
  is_urgent: boolean;
  tags: string[];
  is_anonymized: boolean;
  media: MediaCreate[];
}

export interface MediaCreate {
  name: string;
  type: 'image' | 'video' | 'document' | 'audio';
  mime_type: string;
  url: string;
  size: number;
  width?: number;
  height?: number;
  duration?: number;
}

export interface PostUpdate {
  title?: string;
  summary?: string;
  content?: string;
  specialty?: string;
  is_urgent?: boolean;
  is_resolved?: boolean;
  tags?: string[];
  updated_at?: string;
}

export interface PostResponse {
  success: boolean;
  message: string;
  data?: Post;
}

export interface PostListResponse {
  success: boolean;
  message: string;
  data: Post[];
  total: number;
  page: number;
  limit: number;
}

export interface PostStats {
  total_posts: number;
  urgent_posts: number;
  resolved_posts: number;
  posts_by_specialty: Record<string, number>;
  posts_by_month: Record<string, number>;
}

export interface PostFilters {
  skip?: number;
  limit?: number;
  specialty?: string;
  is_urgent?: boolean;
  is_resolved?: boolean;
  search?: string;
  author_id?: string;
}

File: backend/app/schemas/post.py
from pydantic import BaseModel, Field, validator
from typing import Optional, List
from datetime import datetime
from ..models.base import PyObjectId
from .media import Media, MediaCreate


class PostBase(BaseModel):
    title: str
    summary: Optional[str] = None
    content: Optional[str] = None
    specialty: Optional[str] = None
    is_urgent: bool = False
    tags: Optional[List[str]] = []
    is_anonymized: bool = True

    @validator('title')
    def title_must_not_be_empty(cls, v):
        if not v or not v.strip():
            raise ValueError('Le titre ne peut pas être vide')
        return v.strip()

    @validator('is_anonymized')
    def must_be_anonymized(cls, v):
        if not v:
            raise ValueError('Vous devez confirmer l\'anonymisation des données patients')
        return v


class PostCreate(PostBase):
    media: Optional[List[MediaCreate]] = []


class PostUpdate(BaseModel):
    title: Optional[str] = None
    summary: Optional[str] = None
    content: Optional[str] = None
    specialty: Optional[str] = None
    is_urgent: Optional[bool] = None
    is_resolved: Optional[bool] = None
    tags: Optional[List[str]] = None
    updated_at: Optional[datetime] = Field(default_factory=datetime.utcnow)

    @validator('title')
    def title_must_not_be_empty(cls, v):
        if v is not None and (not v or not v.strip()):
            raise ValueError('Le titre ne peut pas être vide')
        return v.strip() if v else v


class Post(PostBase):
    id: PyObjectId = Field(alias="_id")
    author_id: PyObjectId
    media: Optional[List[Media]] = []
    created_at: datetime
    updated_at: Optional[datetime] = None
    comment_count: int = 0
    like_count: int = 0
    solution_count: int = 0
    is_resolved: bool = False
    ai_suggest: Optional[str] = None

    class Config:
        populate_by_name = True


class PostWithAuthor(Post):
    author: Optional[dict] = None  # Informations de l'auteur populées


class PostResponse(BaseModel):
    success: bool
    message: str
    data: Optional[Post] = None


class PostListResponse(BaseModel):
    success: bool
    message: str
    data: List[Post]
    total: int
    page: int
    limit: int


class PostStats(BaseModel):
    total_posts: int
    urgent_posts: int
    resolved_posts: int
    posts_by_specialty: dict
    posts_by_month: dict
